### DISCUSSION TITLE
passing parameter across livewire components (each on separate view)

### DISCUSSION THREAD

**Discussion by cbhai** [livewire/livewire#3688] [1. Help] [answered]
I have two CRUDS Estimate & EstimateDetail. Project has all usual crud view files. Each view has its own livewire component.

Inside Estimate Create livewire component, I want to do redirect (redirect or any other way go to ) to EstimateDetail Create View ( has second livewire component.

I tried various return statements, but none worked. Cannot pass parameter ( object or just ID). 

Please suggest how can I get ID/object on second livewire component. Any method is OK. Thank you

```
//method inside first livewire component 
public function addRoom(){

$estimate = Estimate::create([
            'title' => $this->title,
            'date' => $this->estimateDate,
            'customer_id' => $this->customerID,
            'total' => 0,
        ]);

return redirect()->route('admin.estimate-details.create', [$estimate]);

return redirect()->route('admin.estimate-details.create', ['estimateID' => $estimate->id]);

return view('admin.estimate-detail.create', compact('estimate'));

}


//EstimateDetailController has method
public function create()
{
        return view('admin.estimate-detail.create');
}

//Route file has
// Estimate Detail
Route::resource('estimate-details', EstimateDetailController::class, ['except' => ['store', 'update', 'destroy']]);

```

**Comment by jonneroelofs** [comment, marked as answer]
I think that in in a livewire component you do not return a redirect but you do:
$this->redirect()

**Reply by cbhai** [reply]
Thank you for your help.
